request_information.go

```diff
@@ -72,23 +72,14 @@ func (request *RequestInformation) GetUri() (*u.URL, error) {
 			return nil, errors.New("pathParameters must contain a value for \"baseurl\" for the url to be built")
 		}
 
-		uriTemplate, err := t.New(request.UrlTemplate)
-		if err != nil {
-			return nil, err
-		}
-		values := t.Values{}
-		varNames := uriTemplate.Varnames()

```

Comment:
Here I'm slightly changing the implementation as it's different from other implementations.

We are not relying on the "extracted varnames" in other implementations, and I do believe that this might become a perf issue.
Please, let me know if I missed any invariant (and how to reproduce in the test-suite).

Reply:
The reason why this was done this way is because a query parameter named "startDateTime" will have a property name of "StartDateTime", and then the substitution key won't match the parameter name, and it will be ignored.

You can find such an example [here](https://github.com/microsoftgraph/msgraph-sdk-go/blob/595a88821dcb211e32da55bcbb7c5194640b0f40/groups/item_calendar_events_delta_request_builder.go#L30) with [this URI template](https://github.com/microsoftgraph/msgraph-sdk-go/blob/595a88821dcb211e32da55bcbb7c5194640b0f40/groups/item_calendar_events_delta_request_builder.go#L46)

But as you can see from the other parameter names, we're able to add an annotation now that will be used. So I believe we should update the code gen to systematically include that annotation. Then this implementation will work.

Just adding a case [here](https://github.com/microsoft/kiota/blob/4e35219011adc5fb9fce42cc1e2a66847d072c89/src/Kiota.Builder/Writers/Go/CodePropertyWriter.cs#L22) when it's not escaped to add the annotation with the name as it is (no normalization) should be fine. Alternatively we could add a refiner to add a serialization name when none is present with the original casing (c.f. your issue about doing all normalization in refiners)